erroranalysis/tests/test_importances.py

```diff
@@ -115,7 +115,11 @@ def test_large_data_importances(self, error_correlation_method):
         print(execution_time)
         # assert we don't take too long and downsample the dataset
         # note execution time is in seconds
-        assert execution_time < 20
+        # note GBM_SHAP is slower than the other methods
+        if error_correlation_method == GBM_SHAP:
+            assert execution_time < 35
+        else:
+            assert execution_time < 25
```

Comment:
I think adding timeouts like these makes the tests very flaky. Is there any other way you validate this? maybe have a higher upper bound of like a minute or so where you know that execution time has deteriorated that it needs to be looked at.

Reply:
These have been working fine previously - the issue is that this bug indeed was due to this code path taking too long.  In reality even this timeout is much longer than what I would usually see, just 1-4 seconds.  But it seems it happens once in a blue moon, and since we have so many builds and tests I just bump up the time to make it happen even less.  I don't want to bump it up too high though because sometimes indeed there can be an issue due to some code change and I want to keep the performance fast.